### ISSUE TITLE
[timeseries] DirectTabular & RecursiveTabular models fail if eval_metric = "WAPE"

### ISSUE THREAD

**Issue by Antoine-Schwartz** [autogluon/autogluon#3927]
**Bug Report Checklist**

- [x] I provided code that demonstrates a minimal reproducible example.  
- [x] I confirmed bug exists on the latest mainline of AutoGluon via source install.  
- [ ] I confirmed bug exists on the latest stable version of AutoGluon.  

**Describe the bug**
If `eval_metric=="WAPE"` in `TimeSeriesPredictor`, DirectTabular & RecursiveTabular fail during training.

**Expected behavior**
Models train normally.

**To Reproduce**
```python
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor
data = TimeSeriesDataFrame("https://autogluon.s3-us-west-2.amazonaws.com/datasets/timeseries/m4_hourly_tiny/train.csv")
predictor = TimeSeriesPredictor(eval_metric="WAPE").fit(data, hyperparameters={"DirectTabular": {}, "RecursiveTabular": {}})
```

**Error message**
```
Models that will be trained: ['RecursiveTabular', 'DirectTabular']
Training timeseries model RecursiveTabular. 
	Warning: Exception caused RecursiveTabular to fail during training... Skipping this model.
	Can't pickle <function mean_absolute_error at 0x7f8fab2b0700>: it's not the same object as sklearn.metrics._regression.mean_absolute_error
Training timeseries model DirectTabular. 
	Warning: Exception caused DirectTabular to fail during training... Skipping this model.
	Can't pickle <function mean_absolute_error at 0x7f8fab2b0700>: it's not the same object as sklearn.metrics._regression.mean_absolute_error
Training complete. Models trained: []
```

**Installed Versions**
AutoGluon v1.0

**Comment by shchur** [comment]
Hi, thank you for reporting the problem. 

Unfortunately, I wasn't able to reproduce it on my machine. In a clean Python 3.10 environment on Linux with AutoGluon installed via `pip install autogluon`, the code that you shared results in both models train without failures.

Can you please share the output of the following command?
```python
from autogluon.core.utils import show_versions
show_versions()
```

**Comment by shchur** [comment]
Are there any additional import statements in your script/notebook, or do you only execute the following 3 lines?
```python
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor
data = TimeSeriesDataFrame("https://autogluon.s3-us-west-2.amazonaws.com/datasets/timeseries/m4_hourly_tiny/train.csv")
predictor = TimeSeriesPredictor(eval_metric="WAPE").fit(data, hyperparameters={"DirectTabular": {}, "RecursiveTabular": {}})
```

**Comment by Antoine-Schwartz** [comment]
I install `autogluon.timeseries==1.0.0` via pip, as well as a private utilities package, whose requirements are : 
```
boto3==1.24.59
pandas==1.3.5
freezegun==1.2.2
pyyaml==5.4.1
uritools==4.0.0
pyarrow==8.0.0
fsspec==2022.8.2
chispa==0.9.2
pydeequ==1.0.1
```
Then I only execute the 3 lines.

**Comment by shchur** [comment]
According to the documentation, [ML Runtime](https://docs.databricks.com/en/release-notes/runtime/14.3lts-ml.html) comes with `scikit-learn-1.1.1`, while the AutoGluon installation installs version `scikit-learn-1.4.1post1` (as you showed here https://github.com/autogluon/autogluon/issues/3927#issuecomment-1953952252). 

I guess that the following happens (but I'm not sure how to verify that this is the source of the problem, or how to fix it):
1. scikit-learn v1.1 gets imported
2. You install autogluon.timeseries, which install scikit-learn v1.4 (probably in a notebook cell with `!pip install autogluon.timeseries`)
3. The predictor cannot pickle the metric during training because scikit-learn version changed

I would recommend trying one of the following:
- restarting the Jupyter notebook after installing autogluon
- running `pip install` outside of the notebook
- executing the code in a `.py` script rather than a notebook

If that doesn't work, I would recommend using a CPU instance. From my experience, `TimeSeriesPredictor` may train even faster on a machine with 32 CPU cores and no GPU compared to a machine with a GPU but fewer CPU cores.

**Comment by Antoine-Schwartz** [comment]
Unfortunately I tried any kind of installation (compute level, notebook level, forcing different versions of sklearn, etc.) and impossible to make the code work with the ML runtime (in notebook).

Otherwise for my experiments with CPU vs GPU, for the deep learning algorithms (this is what works best for my problem), with pytorch and G5 machines on AWS, 32 CPU cores cannot compete at all, it is about 10 times slower (except for RNN-based inference).

**Comment by Antoine-Schwartz** [comment]
No more problems with version 1.2.0 :)

**Comment by MattBoraske** [comment]
@shchur @Antoine-Schwartz Unfortunately, I am encountering this error on databricks despite using autogluon 1.2.0

Since I believe this is an issue with my package install locations on my remote env (databricks), the analysis of my encounter with this error is in issue #3958. @Antoine-Schwartz could you please share your configuration for your databricks environment? I've tried using Databricks 16.0 ML and Standard, and have tried installing autogluon as both a cluster library and locally in a notebook. I've also tried to the same for scikit-learn, but nothing has resolved this error when training the tabular models.

Thanks for your kindness and help in advance!

**Comment by Antoine-Schwartz** [comment]
Hi @MattBoraske,

That's my compute configurations:
Runtime: 14.3 LTS ML
Driver: g5.4xlarge

Installation:
`pip install autogluon.timeseries==1.2.0 deepspeed==0.14.4`
